src/core/trtserver.cc

```diff
@@ -626,6 +626,47 @@ TrtServerResponse::OutputData(
   return nullptr;  // Success
 }
 
+//
+// TrtServerModelIndex
+//
+// Implementation for TRTSERVER2_ModelIndex.
+//
+class TrtServerModelIndex {
+ public:
+  TrtServerModelIndex(const ni::ModelRepositoryIndex& model_repository_index);
+  TRTSERVER_Error* GetModelNames(
+      const char* const** models, uint64_t* models_count);
+
+ private:
+  ni::ModelRepositoryIndex model_repository_index_;
+  std::vector<const char*> index_;
+};
+
+TrtServerModelIndex::TrtServerModelIndex(
+    const ni::ModelRepositoryIndex& model_repository_index)
+    : model_repository_index_(model_repository_index) {
+{
+  for (const auto& model : model_repository_index.models())
```

Comment:
You need to use model_repository_index_ so that your pointer references are into your copy of ModelRepositoryIndex.